### PR TITLE
Adding trace ids to email tags

### PR DESCRIPTION
**Pull request by mayabose**
## Description
Adding DataDog trace and span ids to email tags. Updated `email.test.js` to set default values for the tags.

### Automated and Unit Tests
- [ ] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [ ] Provided ticket and description
- [ ] Provided screenshots/demo
- [ ] Provided testing information
- [ ] Provided adequate test coverage for all new code
- [ ] Added PR reviewers

### CODE REVIEW

**Comment by mayabose** [comment]
Not sure if `email.test.js` is correctly updated? I set the default values to `undefined`, but there's probably some other value I should be initializing it to.

File: packages/server/src/lib/email.js
Comment (TylerHendrickson) [inline packages/server/src/lib/email.js:54]: I think we should conditionally include the `dd_trace_id` and `dd_span_id` email tags so that they are only present when not `undefined`.
```suggestion
    const activeContext = tracer.scope().active()?.context();
    const traceId = activeContext?.toTraceId();
    const spanId = activeContext?.toSpanId();
    const ddTraceTags = [];
    if (traceId) {
        ddTraceTags.push(`dd_trace_id=${traceId}`);
        if (spanId) {
            ddTraceTags.push(`dd_span_id=${spanId}`);
        }
    }
```

File: packages/server/src/lib/email.js
Comment (TylerHendrickson) [inline packages/server/src/lib/email.js:78]: Per the above suggestion:
```suggestion
            ...ddTraceTags,
```

File: packages/server/__tests__/email/email.test.js
Comment (TylerHendrickson) [inline packages/server/__tests__/email/email.test.js:254]: Given the other suggestion regarding conditional inclusion, I think we can leave the tests unmodified, since it would be pretty difficult to instrument tests with dummy trace IDs and span IDs.
```suggestion
                tags: ['notification_type=passcode', 'user_role=staff', 'organization_id=0', 'team_id=0', 'service=test-dd-service', 'env=test-dd-env', 'version=test-dd-version'],
```

File: packages/server/__tests__/email/email.test.js
Comment (TylerHendrickson) [inline packages/server/__tests__/email/email.test.js:273]: (See above)
```suggestion
                tags: ['notification_type=welcome', 'user_role=staff', 'organization_id=0', 'team_id=1', 'service=test-dd-service', 'env=test-dd-env', 'version=test-dd-version'],
```

File: packages/server/__tests__/email/email.test.js
Comment (TylerHendrickson) [inline packages/server/__tests__/email/email.test.js:398]: (See above)
```suggestion
```

File: packages/server/__tests__/email/email.test.js
Comment (TylerHendrickson) [inline packages/server/__tests__/email/email.test.js:421]: (See above)
```suggestion
```

File: packages/server/__tests__/email/email.test.js
Comment (TylerHendrickson) [inline packages/server/__tests__/email/email.test.js:453]: (See above)
```suggestion
```

**Review by TylerHendrickson — CHANGES_REQUESTED** [review]
@mayabose I dropped a number of suggestions here, but they all really come down to the original suggestion that we only include trace IDs and span IDs when they are undefined – since the testing changes were pretty repetitive, I included them as comments.

Also, before it's totally ready, this PR will need to include a new [`trace_id_remapper` processor step](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/logs_custom_pipeline#nestedblock--processor--trace_id_remapper) in the `datadog_logs_custom_pipeline` Terraform resource. Since the rest of that pipeline is defined in #3387, I suggest waiting until that PR is merged, and then updating this branch to amend the pipeline definition.

**Review by TylerHendrickson — APPROVED** [review]
@mayabose Looks good!

**Pull request merged**
